### DISCUSSION TITLE
How to export ifc4 with multilayer walls parameter ?

### DISCUSSION THREAD

**Discussion by voja5** [cwapi3d/cwapi3dpython#130] [Q&A] [answered]
I need to export a file.ifc4 with multilayer walls parameter. In the user interface it is possible but i like to do it with the BIM-Controller.
I suggest to take the function: "export_ifc4_silently_with_options(element_i_ds, file_path, options)" but I am not sure if it's possible. If yes does anynone know how to use the "options"-parameter properly ?

**Comment by Brunner246** [comment, marked as answer]
Hi,

The options for the multilayer wall are not implemented yet. But we can add them. 

General usage example

```python
import cadwork

import bim_controller
import element_controller

ifc_options = bim_controller.get_ifc_options() # or create an empty object =>  cadwork.ifc_options()
ifc_options_lod = ifc_options.get_ifc_options_level_of_detail()
ifc_options_lod.get_cut_drillings()

ifc_options_lod.set_cut_drillings(True)
ifc_options_lod.get_cut_drillings()

ifc_options_lod.set_export_vba_drillings(True)
ifc_options_lod.set_export_vba_components(True)

element_ids = element_controller.get_active_identifiable_element_ids()

bim_controller.export_ifc4_silently_with_options(element_ids, r"C:\Users\Doe\Downloads\test.ifc", ifc_options)
````

**Reply by Brunner246** [reply]
You probably have to create the `ifc_options` from `cadwork.ifc_options` since the other method is may not released yet.
